connectors/src/connectors/google_drive/index.ts

```diff
@@ -92,6 +93,85 @@ export async function createGoogleDriveConnector(
   }
 }
 
+export async function updateGoogleDriveConnector(
+  connectorId: ModelId,
+  {
+    connectionId,
+  }: {
+    connectionId?: NangoConnectionId | null;
+  }
+): Promise<Result<string, ConnectorsAPIErrorResponse>> {
+  if (!NANGO_GOOGLE_DRIVE_CONNECTOR_ID) {
+    throw new Error("NANGO_GOOGLE_DRIVE_CONNECTOR_ID not set");
+  }
+
+  const c = await Connector.findOne({
+    where: {
+      id: connectorId,
+    },
+  });
+  if (!c) {
+    logger.error({ connectorId }, "Connector not found");
+    return new Err({
+      error: {
+        message: "Connector not found",
+        type: "connector_not_found",
+      },
+    } as ConnectorsAPIErrorResponse);
+  }
+
+  // Ideally we want to check that the Google Project ID is the same as the one from the connector
+  // I couln't find an easy way to access it from the googleapis library
+  // Workaround is checking the domain of the user who is updating the connector
+  if (connectionId) {
+    const oldConnectionId = c.connectionId;
+    const currentDriveClient = await getDriveClient(oldConnectionId);
+    const currentDriveUser = await currentDriveClient.about.get({
+      fields: "user",
+    });
+    const currentUserEmail = currentDriveUser.data?.user?.emailAddress || "";
+    const currentDriveUserDomain = currentUserEmail.split("@")[1];
+
+    const newDriveClient = await getDriveClient(connectionId);
+    const newDriveUser = await newDriveClient.about.get({
+      fields: "user",
+    });
+    const newDriveUserEmail = newDriveUser.data?.user?.emailAddress || "";
+    const newDriveUserDomain = newDriveUserEmail.split("@")[1];
+
+    if (!currentDriveUserDomain || !newDriveUserDomain) {
```

Comment:
Do we get the email for all users ?

Reply:
For the two accounts I tested yes 😬 
But indeed I just (finally) found the valid doc and 

About returns this: https://developers.google.com/drive/api/reference/rest/v3/User?hl=fr
And the User has an optional address: https://developers.google.com/drive/api/reference/rest/v3/User?hl=fr

> Uniquement en sortie. Adresse e-mail de l'utilisateur. Elle peut ne pas être présente dans certains cas si l'utilisateur n'a pas rendu son adresse e-mail visible par le demandeur.

If that happens then this specific user will be blocked and won't be able to update the permission. I guess that's okay enough?

Reply:
I see one pontential issue:
if someone signs up via lets say github, I think we never request the scope to get their email in the Google Drive Oauth process. Testing is needed to see what happens in this case.

Reply:
Even if you signed up by Github, if you want to setup a Grive connector you'll have to oAuth with your Google account right? I don't see why we wouldn't get the email this way, can you explain?

Reply:
I just tested this and it works: 
- Create a new workspace attached to my Github account (@gmail.com)
- Create a gdrive datasource for casquedelumiere
- Edit permissions: Try to add another folder for the same gdrive -> works
- Edit permissions: Select dust workspace -> blocked as expected.

Reply:
What do you mean by "Disconnected from Dust"? 🙏🏻